### PR TITLE
filer: propagate lazy metadata deletes to remote mounts

### PR DESCRIPTION
**Pull request by petedodd-pd**
# What problem are we solving?

Apart of #8391.

Deleting filer metadata for objects mounted from remote storage did not remove the corresponding remote object/directory, which could leave stale data in the remote backend and drift between filer metadata and remote state.

# How are we solving the problem?

- Added remote-delete propagation in filer delete flow via `maybeDeleteFromRemote`.
- Before deleting filer metadata, we now:
  - delete remote files for remote-only mounted entries, and
  - remove remote directories for mounted directory entries.
- Remote “not found” is treated as success; other remote delete errors fail the filer delete to avoid inconsistency.
- Added targeted unit tests covering success, not-under-mount, not-found, error, and directory delete paths.

# How is the PR tested?

- Added/updated unit tests in `weed/filer/filer_lazy_remote_test.go` for lazy metadata delete behaviour.

# Checks
- [x] I have added unit tests if possible.
- [ ] I will add related wiki document changes and link to this PR after merging.
- [x] All AI code review comments have been addressed. No more comments to fix if reviewed again. Reviewer may request additional gemini and copilot reviews.

# Checks for AI generated PRs
- [x] I have reviewed every line of code.
- [x] The PR is kept as minimum as possible. Large PRs would not be accepted.

## Release Notes

* **New Features**
  * Introduced remote storage deletion support for files and directories during cleanup operations.
  * Recursive folder deletion now propagates to remote storage with proper error handling.

* **Tests**
  * Added comprehensive test coverage for remote storage deletion scenarios, including error handling and edge cases.

### CODE REVIEW

File: weed/filer/filer.go [hunk 1]
```diff
@@ -151,6 +154,9 @@ func (f *Filer) ListExistingPeerUpdates(ctx context.Context) (existingNodes []*m
 
 func (f *Filer) SetStore(store FilerStore) (isFresh bool) {
 	f.Store = NewFilerStoreWrapper(store)
+	f.remoteDeletionLoop.Do(func() {
```
Comment (chrislusf) [inline weed/filer/filer.go]: It should be started explicitly during filer startup (e.g., in NewFiler or a dedicated
  Start method), not buried as a side effect in a setter gated by sync.Once.
